src/typechecker.erl

```diff
@@ -2436,18 +2448,19 @@ do_type_check_expr_in(Env, ResTy, {record_index, Anno, Name, FieldWithAnno} = Re
     IndexTy = {integer, erl_anno:new(0), Index},
     case subtype(IndexTy, ResTy, Env) of
         {true, Cs} ->
-            {#{}, Cs};
+            {Env, Cs};
         false ->
             throw({type_error, RecIndex, IndexTy, ResTy})
     end;
 
 do_type_check_expr_in(Env, ResTy, {'case', _, Expr, Clauses}) ->
     {ExprTy, VarBinds, Cs1} = type_check_expr(Env, Expr),
-    Env2 = Env#env{ venv = add_var_binds(Env#env.venv, VarBinds, Env) },
-    {VB, Cs2} = check_clauses(Env2, [ExprTy], ResTy, Clauses, capture_vars),
-    {union_var_binds(VarBinds, VB, Env), constraints:combine(Cs1,Cs2)};
+    Env1 = add_var_binds(Env, VarBinds, Env),
+    {Env2, Cs2} = check_clauses(Env1, [ExprTy], ResTy, Clauses, capture_vars),
+    {union_var_binds(VarBinds, Env2, Env), constraints:combine(Cs1,Cs2)};
```

Comment:
Isn't VarBinds included in Env2 already (since it's added to Env1 and Env2 is build from Env1)? So then, `union_var_binds(VarBinds, Env2, Env)` is a no-op. Am I missing something?

Reply:
Indeed, it seems to be redundant :+1: